Redirect Root Route for Datagov Harvest Admin

btylerburton opened the pull request:
# Pull Request

Related to:
- https://github.com/GSA/data.gov/issues/5060
- https://github.com/GSA/data.gov/issues/5080

## About

Also:
- Improves pagination module
- Adds pagination to harvest source summary page
- Drops pagination default to 10 entries

## PR TASKS

- [x] Code well documented
- [x] Tests written, run and passed
- [x] Files linted

btylerburton committed:
updates ui templating;
redirects to all orgs page on load;
udpates harvest source page with feedback changes;
updates fixtures to have better jobs data;
fixes broken tests;
lint

neilmb commented on app/routes.py:
```diff
@@ -474,28 +475,33 @@ def add_harvest_source():
         paginate=False, facets=f"harvest_source_id = '{source.id}'"
     )
-    next_job = "N/A"
+    if len(jobs):
```
**comment (non-blocking)**: I think that the most idiomatic version of this might be `if jobs:`, or maybe `if len(jobs) > 0`, but it's also fine as-is.

neilmb commented on app/routes.py:
```diff
@@ -474,28 +475,33 @@ def add_harvest_source():
     )
-    next_job = "N/A"
+    if len(jobs):
+        last_job = jobs[len(jobs) - 1]
```
**comment (non-blocking)**: More idioms, negative indexes count from the end so `last_job = jobs[-1]` works the same as what you have here.

neilmb commented on app/routes.py:
```diff
@@ -876,13 +882,50 @@ def get_harvest_error(error_id: str = None) -> dict:
+    start_time = current_time - timedelta(hours=24)
+
+    # Format timestamps for SQL query
+    time_filter = f"date_created >= '{start_time.isoformat()}'"
```
**question**: This makes my hackles raise just a tiny bit because it feels like we are interpolating strings into SQL statements, even though it's completely safe in this instance. With pure curiosity, are we pretty sure that `pget_harvest_jobs` is safe from SQL injection?

btylerburton replied:
this is worth having a chat about sanitization and the safety of these functions in general. would appreciate your thoughts on this.

btylerburton replied:
i think i've addressed most of the hackle-raising stuff. it definitely needed some cleanup. as always, thanks for the questions!

neilmb reviewed (dismissed):
LGTM, my comments are just bossy quibbles, nice work.

btylerburton committed:
PR feedback

cmhedrickREI reviewed (dismissed):
I think this looks good! Thanks for fixing the broken test cases!

btylerburton committed:
fixes issues with metrics pagination;
cleans up metrics route code;
fixes issue with pagination numbers in pagination module;
makes pagination per_page count a kwarg;
removes sort functionality from tables as it breaks with htmx updates and is not super useful

btylerburton committed:
corrects test fixed by upstream pr;
calculates count using new count decorator

btylerburton committed:
cleans up pagination; adds db_current property

btylerburton committed:
adds pagination to harvest source summary page;

btylerburton committed:
fix package lock

btylerburton committed:
fix gulp task order

btylerburton committed:
copy update

btylerburton committed:
fix records count call

btylerburton committed:
cleanup fixes

btylerburton committed:
dont show jobs in future

btylerburton committed:
change calculations for harvest source page to only pull latest records

cmhedrickREI reviewed (approved):
I think this all looks good! :rocket:
